lib/public/views/Logs/FilePreview/filePreviewComponent.js

```diff
@@ -0,0 +1,299 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+
+import { StatefulComponent } from '../../../components/common/StatefulComponent.js';
+import spinner from '../../../components/common/spinner.js';
+import errorAlert from '../../../components/common/errorAlert.js';
+import { h, RemoteData } from '/js/src/index.js';
+
+/**
+ * Base class of file display
+ */
+class FileDisplayComponent extends StatefulComponent {
+    /**
+     * Constructor
+     *
+     * @param {Blob|File} file the file or blob to display
+     * @param {string} fileName the display name of the file
+     * @return {Component} the resulting component
+     */
+    oninit({ attrs: { file, fileName } }) {
+        this._fileName = fileName;
+
+        /**
+         * @type {RemoteData<string, ApiError>}
+         * @private
+         */
+        this._remoteFileReadResult = RemoteData.loading();
```

Comment:
I think is should be set with `_setRemoteFileBlob` in order to utilise notify

Reply:
BTW, I don't whether it's intended, but a big spinner appears just on top of the page, for me it would be more intuitive, that it appears inside pop-up holding preview content

Reply:
it makes no sense to notify here, because the render will occur after.

Nice catch for the spinner, I fixed it